### ISSUE TITLE
Associate torrent files and magnet links

### ISSUE THREAD

**Issue by jsieler** [picotorrent/picotorrent#185]
Would be nice to have an option, to associate torrent files and magnet links with picotorrent by default. Opening torrent files from HDD/SSD by downloading them first makes no fun. Copy-paste magnet links also too.

**Comment by vktr** [comment]
If you install PicoTorrent (via the installer bundle, `PicoTorrent-0.7.0-x64.exe`) you will get file type associations and magnet URI handlers. Is this not working for you?

**Comment by jsieler** [comment]
Sorry, i'm using ZIP'ped version.

But in case, if i install other Torrent client, and mess the association, how do i reassociate it again? Just by reinstalling?

**Comment by vktr** [comment]
Yes. PicoTorrent will not reclaim the associations if they are lost - to do that you need to either reinstall or repair the installation.

**Comment by jsieler** [comment]
Any way possible to add a check tick in options where you can re-set it again without reinstalling?

**Comment by jsieler** [comment]
Like this one?
http://share.pride-it.net/v/85fff1f1

**Comment by vktr** [comment]
Sure, sounds reasonable! :smile:

**Comment by Tom4tot** [comment]
In case people were wondering, you can set the portable version of PicoTorrent as the default torrent client following this:
For .torrent files, just right click on a file, open with, tick "always use this app" and then pick the right .exe file.

For magnet, it's a little more complicated but not that much. 
Open regedit, browse here: 
`HKEY_CLASSES_ROOT\Magnet\shell\open\command`
Then double-click the default location and change the data section to:
`"*path to your your .exe*" "%1"`

Maybe that will help you as well @vktr 

Btw, I found the procedure here: https://superuser.com/questions/845135/how-to-change-default-program-to-open-magnet-link-in-windows

**Comment by TechnoGreg** [comment]
I think this is a great  option to come :)

**Comment by vonder2** [comment]
> 
> 
> In case people were wondering, you can set the portable version of PicoTorrent as the default torrent client following this:
> For .torrent files, just right click on a file, open with, tick "always use this app" and then pick the right .exe file.
> 
> For magnet, it's a little more complicated but not that much.
> Open regedit, browse here:
> `HKEY_CLASSES_ROOT\Magnet\shell\open\command`
> Then double-click the default location and change the data section to:
> `"*path to your your .exe*" "%1"`
> 
> Maybe that will help you as well @vktr
> 
> Btw, I found the procedure here: https://superuser.com/questions/845135/how-to-change-default-program-to-open-magnet-link-in-windows

This method not working for Mozilla Firefox...

**Comment by tomsotte** [comment]
scoop does this with a little regedit patch https://github.com/ScoopInstaller/Extras/blob/59b802f3e6b9b9b52c3c632d449afb48f54f023c/scripts/picotorrent/install-associations.reg
